### PR TITLE
Update gradle, agp, kotlin to lastest versions

### PR DESCRIPTION
**Pull request by rougsig**
**Background**
I want try update android gradle plugin and gradle to newest release version

**Changes**
- Up Kotlin from `1.4.0` to `1.5.21`
- Up Gradle from `6.1.1` to `7.0.2`
- Up AGP from `4.0.1` to `7.0.1`
- Up JDK from `1.8` to `1.11`
- Up Dagger from `2.28.3` to `2.38.1`
- Up Moxy from `2.1.2` to `2.2.2`

**Test plan**
- Build project

### CODE REVIEW

File: buildSrc/build.gradle.kts [hunk 1]
```diff
@@ -9,6 +10,11 @@ repositories {
...
-    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.0")
-    implementation("com.android.tools.build:gradle:4.0.1")
+    // SEE NOTE_CONFIGURATION_PLUGIN
+    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.21")
+    implementation("com.android.tools.build:gradle:7.0.1")
 }
+
+// NOTE_CONFIGURATION_PLUGIN
+// when updating this dependency version, also update them in buildSrc/build.gradle.kts
+// (manual update is required, because this file is not seen by build scripts in buildSrc)
```
Comment (LionZXY) [inline buildSrc/build.gradle.kts:20]: What you see about move in `buildSrc/src/main/java/Dependencies.kt` file?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,7 @@
-#Mon Aug 17 21:57:27 EEST 2020
+#Mon Aug 23 14:00:45 EET 2021
 distributionBase=GRADLE_USER_HOME
+#SEE NOTE_CONFIGURATION_PLUGIN
```
Comment (LionZXY) [inline gradle/wrapper/gradle-wrapper.properties:3]: Hm... Are you sure that this is good idea add some comments in autogenerated file?

Comment (rougsig) [reply gradle/wrapper/gradle-wrapper.properties:3]: I think that everything is fine, for the reason that we can see this change in the code review and not forget about this problem.

**Review by LionZXY — APPROVED** [review]
LGTM

**Pull request merged**